infra/src/main/resources/db/migration/repeatable/R__04.06_track_number_in_layout_context.sql

```diff
@@ -35,14 +35,19 @@ select
   where case publication_state_in
           when 'OFFICIAL' then not row.draft
           else row.draft
-            or not exists(select *
-                            from layout.track_number overriding_draft
-                            where overriding_draft.design_id is not distinct from design_id_in
-                              and overriding_draft.draft
-                              and row.id = case
-                                             when row.design_id is null then overriding_draft.official_row_id
-                                             else overriding_draft.design_row_id
-                                           end)
+            or case
+                 when row.design_id is null then
+                   not exists(select *
+                                from layout.track_number overriding_draft
+                                where overriding_draft.design_id is not distinct from design_id_in
+                                  and overriding_draft.draft
+                                  and overriding_draft.official_row_id = row.id)
+                 else not exists(select *
+                                   from layout.track_number overriding_draft
+                                   where overriding_draft.design_id is not distinct from design_id_in
+                                     and overriding_draft.draft
+                                     and overriding_draft.design_row_id = row.id)
+               end
```

Comment:
Erinomaisen ärsyttävältä näyttävä puhdas optimointi. Tässä ei ole minkäänlaista funktionaalista muutosta: Ainoa muutos on, että "row.design_id is null"-ehto on vedetty ylemmälle tasolle. Piti tehdä perffisyistä, koska postgres ei osaa kyseistä optimointia tehdä itse, ja joutui sen takia tekemään overriding_draftien etsimisen koko taulun läpi menevänä seq scannina (jokaista tarkastettavaa riviä kohden).